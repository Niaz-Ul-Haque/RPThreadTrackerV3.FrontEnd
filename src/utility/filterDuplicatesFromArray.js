const filterDuplicatesFromArray = (array) => {
	return array.filter((elem, pos, arr) => elem && arr.indexOf(elem) == pos);
}
export default filterDuplicatesFromArray;
