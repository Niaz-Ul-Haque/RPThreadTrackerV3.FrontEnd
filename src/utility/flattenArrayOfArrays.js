const flattenArrayOfArrays = array => array.reduce((a, b) => a.concat(
	Array.isArray(b) ? flattenArrayOfArrays(b) : b
), []);
export default flattenArrayOfArrays;
