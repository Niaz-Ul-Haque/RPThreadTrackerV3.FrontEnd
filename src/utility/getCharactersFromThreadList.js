import { filterDuplicatesFromArray } from './';

const getCharactersFromThreadList = (threads) => {
	if (!threads || threads.length < 1) {
		return [];
	}
	const characters = threads.map(t => t.thread.character);
	return filterDuplicatesFromArray(characters, 'characterId');
};
export default getCharactersFromThreadList;
