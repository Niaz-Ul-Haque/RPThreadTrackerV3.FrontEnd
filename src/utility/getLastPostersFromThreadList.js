import { filterDuplicatesFromArray } from './';

const getLastPostersFromThreadList = (threads) => {
	if (!threads || threads.length < 1) {
		return [];
	}
	const lastPosters = threads.map((t) => {
		if (!t.status) {
			return '';
		}
		return t.status.LastPosterUrlIdentifier;
	});
	return filterDuplicatesFromArray(lastPosters);
};
export default getLastPostersFromThreadList;
