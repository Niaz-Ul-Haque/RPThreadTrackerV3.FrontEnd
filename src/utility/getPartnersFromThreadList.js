import { filterDuplicatesFromArray } from './';

const getPartnersFromThreadList = (threads) => {
	if (!threads || threads.length < 1) {
		return [];
	}
	const partners = threads.map((t) => {
		if (!t.thread) {
			return '';
		}
		return t.thread.partnerUrlIdentifier;
	});
	return filterDuplicatesFromArray(partners);
};
export default getPartnersFromThreadList;
