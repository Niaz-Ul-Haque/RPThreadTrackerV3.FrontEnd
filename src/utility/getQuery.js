import queryString from 'query-string';

const getQuery = () => {
	try {
		if (typeof (window) !== 'undefined') {
			return queryString.parse(window.location.search);
		}
		return {};
	} catch (e) {
		console.log(e);
		return null;
	}
};
export default getQuery;
