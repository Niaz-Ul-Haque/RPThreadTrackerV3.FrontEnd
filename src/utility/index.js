export { default as getQuery } from './getQuery';
export { default as flattenArrayOfArrays } from './flattenArrayOfArrays';
export { default as filterDuplicatesFromArray } from './filterDuplicatesFromArray';
export { default as getCharactersFromThreadList } from './getCharactersFromThreadList';
export { default as getPartnersFromThreadList } from './getPartnersFromThreadList';
export { default as getLastPostersFromThreadList } from './getLastPostersFromThreadList';
export { default as getTagsFromThreadList } from './getTagsFromThreadList';
