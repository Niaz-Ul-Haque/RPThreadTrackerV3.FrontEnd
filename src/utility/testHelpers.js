export const DATA_SPEC_ATTRIBUTE_NAME = 'data-spec';

export const getSpecWrapper = (componentWrapper, specName) => {
	const specWrappers = componentWrapper.find(`[${DATA_SPEC_ATTRIBUTE_NAME}="${specName}"]`);
	return specWrappers;
};
