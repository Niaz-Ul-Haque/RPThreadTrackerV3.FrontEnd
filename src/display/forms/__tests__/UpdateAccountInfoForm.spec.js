// #region imports
import React from 'react';
import { shallow } from 'enzyme';
import { getSpecWrapper } from '../../../utility/testHelpers';
import UpdateAccountInfoForm from '../update-account-info/UpdateAccountInfoForm';
// #endregion imports

const createTestProps = propOverrides => ({
	handleInputChange: jest.fn(),
	user: {
		userName: 'TestUser',
		email: 'test@test.com'
	},
	...propOverrides
});

describe('rendering', () => {
	it('should render valid snapshot', () => {
		const props = createTestProps();
		const jsx = (<UpdateAccountInfoForm {...props} />);
		const element = shallow(jsx);
		expect(element).toMatchSnapshot();
	});
	it('should populate existing username', () => {
		const props = createTestProps();
		const jsx = (<UpdateAccountInfoForm {...props} />);
		const element = shallow(jsx);
		const field = getSpecWrapper(element, 'username-field');
		expect(field).toHaveValue('TestUser');
	});
	it('should populate existing email and disable field', () => {
		const props = createTestProps();
		const jsx = (<UpdateAccountInfoForm {...props} />);
		const element = shallow(jsx);
		const field = getSpecWrapper(element, 'email-field');
		expect(field).toHaveValue('test@test.com');
		expect(field).toBeDisabled();
	});
	it('should validate the username field', () => {
		const props = createTestProps();
		const jsx = (<UpdateAccountInfoForm {...props} />);
		const element = shallow(jsx);
		const field = getSpecWrapper(element, 'username-field');
		expect(field.props().validate.required).toHaveProperty('value', true);
		expect(field.props().validate.minLength).toHaveProperty('value', 3);
		expect(field.props().validate.maxLength).toHaveProperty('value', 256);
	});
	it('should validate the email field', () => {
		const props = createTestProps();
		const jsx = (<UpdateAccountInfoForm {...props} />);
		const element = shallow(jsx);
		const field = getSpecWrapper(element, 'email-field');
		expect(field.props().validate.required).toHaveProperty('value', true);
		expect(field.props().validate.email).toHaveProperty('value', true);
	});
});

describe('behavior', () => {
	it('should handle input change for username', () => {
		const handleInputChange = jest.fn();
		const props = createTestProps({ handleInputChange });
		const jsx = (<UpdateAccountInfoForm {...props} />);
		const element = shallow(jsx);
		const field = getSpecWrapper(element, 'username-field');
		field.simulate('change');
		expect(handleInputChange).toHaveBeenCalledTimes(1);
	});
});
