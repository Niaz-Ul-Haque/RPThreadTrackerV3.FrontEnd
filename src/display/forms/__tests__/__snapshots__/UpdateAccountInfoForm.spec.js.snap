// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering snapshots should render valid snapshot 1`] = `
<div>
  <FormGroup
    row={true}
    tag="div"
  >
    <Col
      lg="3"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
      xs="12"
    >
      <Label
        htmlFor="username"
        tag="label"
      >
        Username:
      </Label>
    </Col>
    <Col
      lg="9"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
      xs="12"
    >
      <AvField
        data-spec="username-field"
        name="username"
        onChange={[MockFunction]}
        placeholder="Username"
        type="text"
        validate={
          Object {
            "maxLength": Object {
              "errorMessage": "Your username is too long.",
              "value": 256,
            },
            "minLength": Object {
              "errorMessage": "Your username must be more than 3 characters.",
              "value": 3,
            },
            "required": Object {
              "errorMessage": "You must enter a username.",
              "value": true,
            },
          }
        }
        value="TestUser"
      />
    </Col>
  </FormGroup>
  <FormGroup
    row={true}
    tag="div"
  >
    <Col
      lg="3"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
      xs="12"
    >
      <Label
        htmlFor="email"
        tag="label"
      >
        Email:
      </Label>
    </Col>
    <Col
      lg="9"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
      xs="12"
    >
      <AvField
        data-spec="email-field"
        disabled={true}
        name="email"
        onChange={[MockFunction]}
        placeholder="Email"
        type="email"
        validate={
          Object {
            "email": Object {
              "errorMessage": "Please enter a valid email",
              "value": true,
            },
            "required": Object {
              "errorMessage": "You must enter an email.",
              "value": true,
            },
          }
        }
        value="test@test.com"
      />
    </Col>
  </FormGroup>
</div>
`;
